src/architecture/serialization.md

```diff
@@ -1,533 +1,193 @@
-as a [structure](#structures) with two `u32` fields.
-
-### Structures
-
-A structure is representation of [`struct` in Rust][rust_structs]. It is data
-structure with a fixed number of possibly heterogeneous, ordered, named fields.
-
-In binary representation structure is split into two main parts (which are
-adjacent to each other for each serialized structure):
-
-- **Header** is a fixed sized part.
-- **Body** is a dynamic sized part; it can be read only after parsing the header.
-
-Fixed-length fields are stored completely in the header.
-Var-length fields are allocated as segments in the body,
-plus take 8 bytes for the serialized segment pointer, as described [above](#segment-pointers).
-
-Thus, a segment pointer in the header (the position of which is known in compile
-time) points to the segment in the body,
-which contains the actual serialization of the field. Segments are placed
-in the correspondence with [the validation rules](#segment-validation-rules).
-
-!!! note
-    A field of structure type is always handled as var-length field (even if all
-    its fields are fixed-length): segment pointer in the header of the outer
-    structure points to the segment with the inner structure itself.
-
-!!! note "Example"
-    Consider a structure containing `PublicKey`, `u64` and `bool` fields. In the
-    binary format all fields of such structure are placed in the header, its body
-    is empty. So such a structure is fixed-length.
-
-#### Structure composition
-
-`encoding_struct!` macro allows to compose existing structures:
-
-```rust
-encoding_struct! {
-    struct Pair {
-        first: u32,
-        second: u32,
-    }
-}
-
-encoding_struct! {
-    struct Pairs {
-        inner: Vec<Pair>,
-    }
+the network between nodes turns into messages.
+
+**Communication with Light Clients** Light clients form
+[messages](transactions.md) which include
+transactions serialized in protobuf, sign them and send to the network.
+
+**Storage of Data** [The storage](../architecture/storage.md) is used to place
+blocks, configurations, data specific for services. Data obtained from the
+storage is not validated, since it is assumed to be validated earlier.
+
+## Principles of Using Protobuf Serialization
+
+To apply protobuf serialization to structures in Exonum, users are required to
+describe the structures in a file with `.proto` extension, using the
+[protobuf interface description language][language].  All the `.proto` files
+are then combined into a single module which is used to generate files with
+`.rs` extension. The `.rs` files handle the serialization of structures
+described in them. And that is it, the described structures can then be
+serialized and deserialized in Exonum.  
+
+For example, a simple `Timestamp` structure will have following description
+in a `.proto` file:
+
+```protobuf
+message Timestamp {
+
+  // Represents seconds of UTC time since Unix epoch
+  // 1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to
```

Comment:
I find this description confusing because values `0001-01-01T00:00:00Z` don't look like `int64` values.

Reply:
Thats comment from google/well_known_types. I think to show this point comments can be removed.